server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -99,4 +100,25 @@
         return map;
     }
 
+    /**
+     * Returns {@code true} if the two specified maps are equal to one another. Two maps are considered equal if both contain
+     * the same number of entries, and all values ​​of the corresponding keys are equal by the specified equivalence relationship
+     * The primary use case is to check if two maps with array values are equal.
+     *
+     * @param left             one map to be tested for equality
+     * @param right            the other map to be tested for equality
+     * @param valueEquivalence the equivalence relationship for comparing values
+     * @return {@code true} if the two maps are equal
+     */
+    public static <K, V> boolean equals(Map<K, V> left, Map<K, V> right, BiPredicate<V, V> valueEquivalence) {
```

Comment:
Given the use cases for this method, I wonder if it would make more sense to rename it to `deepEquals`, and drop the `BiPredicate` parameter in favour of `Objects.deepEquals`.
Then it will always do the right thing for the use cases we have in mind, and be named consistently with existing JDK methods of a similar purpose.

What do you think?

Reply:
Totally agree on that.